execution-engine/src/fs.rs

```diff
@@ -1918,6 +1918,34 @@ impl FileSystem {
         debug_assert_eq!(read_len, vec.len());
         Ok(vec)
     }
+
+    /// Return whether the given path can be executed by the given principal.
+    /// Fails if the principal can't access the path with `path_open()`.
+    pub fn is_file_executable<T: AsRef<Path> + Copy>(
```

Comment:
Why the `Copy` constraint?  Isn't that very restrictive in terms of the types that actually satisfy it?

Reply:
The borrow checker suggested that. I haven't looked into alternative solutions

Reply:
`Copy` is very restrictive, mainly the primitive types implement it.  `Clone` may be a better alternative.